modules/bizzclickBidAdapter.js

```diff
@@ -1,322 +1,90 @@
-import {_map, deepAccess, deepSetValue, getDNT, logMessage, logWarn} from '../src/utils.js';
-import {registerBidder} from '../src/adapters/bidderFactory.js';
-import {BANNER, NATIVE, VIDEO} from '../src/mediaTypes.js';
-import {config} from '../src/config.js';
-import {convertOrtbRequestToProprietaryNative} from '../src/native.js';
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { deepSetValue } from '../src/utils.js';
 
 const BIDDER_CODE = 'bizzclick';
-const ACCOUNTID_MACROS = '[account_id]';
-const URL_ENDPOINT = `https://us-e-node1.bizzclick.com/bid?rtb_seat_id=prebidjs&secret_key=${ACCOUNTID_MACROS}`;
-const NATIVE_ASSET_IDS = { 0: 'title', 2: 'icon', 3: 'image', 5: 'sponsoredBy', 4: 'body', 1: 'cta' };
-const NATIVE_PARAMS = {
-  title: {
-    id: 0,
-    name: 'title'
-  },
-  icon: {
-    id: 2,
-    type: 1,
-    name: 'img'
-  },
-  image: {
-    id: 3,
-    type: 3,
-    name: 'img'
+const SOURCE_ID_MACRO = '[sourceid]';
+const ACCOUNT_ID_MACRO = '[accountid]';
+const HOST_MACRO = '[host]';
+const URL = `https://${HOST_MACRO}.bizzclick.com/bid?rtb_seat_id=${SOURCE_ID_MACRO}&secret_key=${ACCOUNT_ID_MACRO}&integration_type=prebidjs`;
+const DEFAULT_CURRENCY = 'USD';
+const DEFAULT_TMAX = 500;
+const DEFAULT_HOST = 'us-e-node1';
+
+const converter = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 20,
   },
-  sponsoredBy: {
-    id: 5,
-    name: 'data',
-    type: 1
+  imp(buildImp, bidRequest, context) {
+    const imp = buildImp(bidRequest, context);
+    if (!imp.bidfloor) imp.bidfloor = bidRequest.params.bidfloor || 0;
+    imp.ext = {
+      [BIDDER_CODE]: {
+        accountId: bidRequest.params.accountId,
+        sourceId: bidRequest.params.sourceId,
+        host: bidRequest.params.host || DEFAULT_HOST,
+      }
+    }
+    return imp;
   },
-  body: {
-    id: 4,
-    name: 'data',
-    type: 2
+  request(buildRequest, imps, bidderRequest, context) {
+    const request = buildRequest(imps, bidderRequest, context);
+    const bid = context.bidRequests[0];
+    request.test = config.getConfig('debug') ? 1 : 0;
+    request.tmax = request.tmax || DEFAULT_TMAX;
+    if (!request.cur) request.cur = [bid.params.currency || DEFAULT_CURRENCY];
+
+    if (config.getConfig('coppa')) deepSetValue(request, 'regs.coppa', 1);
```

Comment:
Heads up: most of what's here in `request` is done by ortbConverter:

 - all the consent fields (coppa, usp, gdpr, as well as gpp) are filled in `bidderRequest.ortb2` and automatically copied into the outgoing request as long the relevant modules are installed.  If you want to unit test the base logic from your suite, it can import those modules and use `syncAddFPDToBidderRequest` - [here's an example for reference](https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/improvedigitalBidAdapter_spec.js)
 - `tmax` is set to the bidder timeout configured by the pub, so your fallback here is never used - and IMO doesn't make sense (the request is timed out according to the bidder timeout regardless of what is sent to your exchange)

Reply:
Make sense, fixed